Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp

```diff
@@ -230,14 +230,9 @@ void* OSAllocator::reserveAndCommit(size_t bytes, Usage usage, bool writable, bo
 void OSAllocator::commit(void* address, size_t bytes, bool writable, bool executable)
 {
 #if OS(LINUX)
-    int protection = PROT_READ;
-    if (writable)
-        protection |= PROT_WRITE;
-    if (executable)
-        protection |= PROT_EXEC;
-    if (mprotect(address, bytes, protection))
-        CRASH();
-    madvise(address, bytes, MADV_WILLNEED);
+    UNUSED_PARAM(writable);
+    UNUSED_PARAM(executable);
+    while (madvise(address, bytes, MADV_WILLNEED) == -1 && errno == EAGAIN) { }
```

Comment:
Given that the `writable` and `executable` parameters were only used under `OS(LINUX)`, I think it would make sense to remove the parameters completely, and apply the same change to `OSAllocatorWin.cpp`—unless there is some reason why that would be an issue on Windows, of course.

Reply:
That sounds good! I'll do it in a follow-up patch :)